x-pack/filebeat/module/aws/elb/ingest/pipeline.yml

```diff
@@ -0,0 +1,151 @@
+description: "Pipeline for ELB logs"
+
+processors:
+  - grok:
+      field: message
+      # Classic ELB patterns documented in https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/access-log-collection.html
+      # ELB v2 Application load balancers https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html
+      # ELB v2 Netwwork load balancers https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-access-logs.html
+      #
+      patterns:
+        # HTTP (Classic ELB)
+        - >-
+          %{ELBHTTPLOG}
+
+        # TCP (Classic ELB)
+        - >-
+          %{ELBTCPLOG}
+
+        # HTTP from Application Load Balancers (v2 Load Balancers)
```

Comment:
Thanks for adding both ALB and NLB here. One question, for metricbeat, I'm thinking to separate elb/alb/nlb into different metricsets just because the namespace is different. I'm not suggesting we have to separate these three into different filesets here. But will this cause a problem because of the difference between fileset and metricset?

Reply:
Good question, I though about separating them in different filesets because they actually have different formats, but at the end I decided to keep them in an only fileset, main reasons:
* Log-wise they are not so different at the end and lots of patters can be reused (this may be different with metricbeat).
* Product-wise I find the separation of load balancers a bit blurry. All load balancers are created from the same UI, and there is a migration wizard from ELB to the new Load Balancers. Probably loads of users still use the classic load balancers, and use the new ones only because they are the current defaults. So I though that if possible, it'd be better to support all the options from a single fileset, so users don't need to think about which one to use for the load balancers they have or plan having.

It'd be nice if for metricbeat we could also support all the load balancers from same metricset, but it maybe something harder to achieve if their metrics are very different.

Reply:
Good point, I will look into how we can make elb metricset to collect from different namespaces 👍